### PR TITLE
Added Binary Search [C++]

### LINKED ISSUE #548: Binary Search [C++]
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS).

 Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by ishakalra**
Fixes #548

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by singhpratyush** [comment]
@ishakalra: Please fix travis build. Thanks :)

**Comment by ishakalra** [comment]
@singhpratyush Hi, earlier the build failed due to indent spaces/tabs, I tried modifying that but now I can't seem to figure out where the fault is. Would be really grateful if you could guide me through the issue!

File: binary_search/binarysearch.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include<iostream>
+using namespace std;
```
Comment (mohitkyadav) [inline binary_search/binarysearch.cpp]: Give a blank line at line 2 and 3.

File: binary_search/binarysearch.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+int binarySearchIterative(int array[], int start, int end, int find)
+{
```
Comment (mohitkyadav) [inline binary_search/binarysearch.cpp]: Please follow Google's C++ style guide. https://google.github.io/styleguide/cppguide.html

File: binary_search/binarysearch.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	while(end >= start)
+	{
+		int mid = start + (end - start)/2;
```
Comment (mohitkyadav) [inline binary_search/binarysearch.cpp]: Put space around binary operators.

File: binary_search/binarysearch.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		if(array[mid] == find)
+			return mid;
+		else if(array[mid] < find)
+			start = mid + 1;
+		else
+			end = mid - 1;
+	}
+	return -1;
+}
+int binarySearchRecursive(int array[], int start, int end, int find)
```
Comment (mohitkyadav) [inline binary_search/binarysearch.cpp]: Blank line at line 17.

File: binary_search/binarysearch.cpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+			return binarySearchRecursive(array, mid + 1, end, find);
+	}
+	return -1;
+}
+int main()
+{
+	int array[] = { 1, 2, 3, 4, 5, 6, 11, 23, 25, 66, 88, 99, 113 };
+	int end = sizeof(array)/sizeof(array[0]);
+	cout<<binarySearchIterative(array,0,end,1)<<'\n'<<binarySearchRecursive(array,0,end,1)<<'\n';
+	cout<<binarySearchIterative(array,0,end,0)<<'\n'<<binarySearchRecursive(array,0,end,0)<<'\n';
```
Comment (mohitkyadav) [inline binary_search/binarysearch.cpp]: Put space around `<<`.

File: binary_search/binarysearch.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	return 1;
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: @ishakalra Try returning `0` here, let's see if the build is fixed.

File: binary_search/binarysearch.cpp [hunk 7]
```diff
@@ -0,0 +1,38 @@
+#include<iostream>
+using namespace std;
+int binarySearchIterative(int array[], int start, int end, int find)
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: Use underscores instead of `camelCase`.

Comment (Monal5031) [inline binary_search/binarysearch.cpp]: Use underscores for naming variables and functions instead of `camelCase` in whole code.

**Comment by singhpratyush** [comment]
@ishakalra: The indentation is good now. The issue is due to non zero return code by the `main` method. Travis build would be fixed after making it return 0.

**Comment by singhpratyush** [comment]
The build is now fixed :tada:. Please take a look at comments from @mohitkyadav and @Monal5031 and make if needed.

File: binary_search/binarysearch.cpp [hunk 8]
```diff
@@ -0,0 +1,42 @@
+#include<iostream>
+
+using namespace std;
+
+int binary_search_iterative(int array[], int start, int end, int find)
+{
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: The indentation is not according to standard cpp guidelines.
The opening curly brace always starts in same line.
`int name_of_func(...) {` Same changes in rest of the file.

File: binary_search/binarysearch.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+	while(end >= start)
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: Please make this, `while (...) {` as per standard cpp guidelines. Same changes in rest of the file.

File: binary_search/binarysearch.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+	{
+		int mid = start + (end - start)/2;
+		if(array[mid] == find)
+			return mid;
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: Add whitespace around binary operator `/`. Same changes in rest of the file.

File: binary_search/binarysearch.cpp [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+		else if(array[mid] < find)
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: `else if (...) {` Same changes in rest of the file.

File: binary_search/binarysearch.cpp [hunk 12]
```diff
@@ -0,0 +1,42 @@
+#include<iostream>
+
+using namespace std;
+
+int binary_search_iterative(int array[], int start, int end, int find)
+{
+	while(end >= start)
+	{
+		int mid = start + (end - start)/2;
+		if(array[mid] == find)
```
Comment (Monal5031) [inline binary_search/binarysearch.cpp]: `if (...) {` Same changes in rest of the file.

**Pull request closed without merging**